app/src/main/java/com/iti/android/team1/ebuy/ui/register_screen/view/RegisterScreen.kt

```diff
@@ -1,39 +1,121 @@
+import com.iti.android.team1.ebuy.ui.register_screen.RegisterResult
+import com.iti.android.team1.ebuy.ui.register_screen.viewmodel.RegisterViewModel
+import com.iti.android.team1.ebuy.ui.register_screen.viewmodel.RegisterViewModelFactory
 
 class RegisterScreen : Fragment() {
 
-    companion object {
-        fun newInstance() = RegisterScreen()
+
+    private var _binding: FragmentRegisterScreenBinding? = null
+
+
+    val viewModel: RegisterViewModel by viewModels {
+        RegisterViewModelFactory(Repository(LocalSource(requireContext())))
     }
 
-    private var _binding:FragmentRegisterScreenBinding? = null
-    private lateinit var viewModel: RegisterScreenViewModel
     private val binding get() = _binding!!
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
+        savedInstanceState: Bundle?,
     ): View {
-        _binding = FragmentRegisterScreenBinding.inflate(inflater,container,false)
+        _binding = FragmentRegisterScreenBinding.inflate(inflater, container, false)
         return binding.root
     }
 
 
+    @RequiresApi(Build.VERSION_CODES.O)
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-            binding.btnSignIn.setOnClickListener {
-                findNavController().popBackStack()
+        binding.btnSignIn.setOnClickListener {
+            findNavController().popBackStack()
+        }
+        binding.btnSignUp.setOnClickListener {
+            viewModel.registerCustomer(collectDataFromFields())
+        }
+
+        viewModel.registerLiveData.observe(viewLifecycleOwner) {
+            when (it) {
+                is RegisterResult.Loading -> showProgressBar()
+
+                is RegisterResult.InvalidData -> {
+                    showWrongInputResult(it.error)
+                    hideProgressBar()
+                }
+
+                is RegisterResult.RegisterFail -> {
+                    Toast.makeText(requireContext(),
+                        "fail ${it.errorMsg}",
+                        Toast.LENGTH_SHORT).show()
+                    Log.i("TAG", "onViewCreated:${it.errorMsg} ")
+                    hideProgressBar()
+                }
+                is RegisterResult.RegisterSuccess -> {
+                    Toast.makeText(requireContext(),
+                        "Welcome ${it.customer.firstName}",
+                        Toast.LENGTH_SHORT).show()
+                    hideProgressBar()
+                }
             }
+        }
+
+    }
+
+    private fun showProgressBar() {
+        binding.progress.visibility = View.VISIBLE
+        binding.btnSignUp.isEnabled = false
+        binding.textInputLayout.isEnabled = false
+        binding.textInputLayout2.isEnabled = false
+        binding.textInputLayout3.isEnabled = false
+        binding.textInputLayout4.isEnabled = false
```

Comment:
Instead of this boilerplate code, you could place them in constraint layout and disable the root only.

Reply:
i tried but failed so i have no way